docs/nx-cloud/intro/why-nx-cloud.md

```diff
@@ -0,0 +1,64 @@
+# Why Nx Cloud?
+
+{% youtube
+src="https://youtu.be/4VI-q943J3o"
+title="Fast CI for monorepos"
+width="100%" /%}
+
+If CI feels challenging to you, **it might not be your fault**. It's a fundamental issue with how the current, traditional CI execution model works. Nx Cloud adopts a new **task-based** CI model which allows to overcome the limitations of slowness but also unreliability of the current VM-based CI execution. _(Dive deeper into the task based CI execution model [in this blog post](https://blog.nrwl.io/reliable-ci-a-new-execution-model-fixing-both-flakiness-and-slowness-6849fd4b4037))_
+
+Such approach has led to great results:
+
+- 30 - 70% faster CI
+- 40 - 75% reduction in CI costs
+- 9x faster e2e test execution
+- robust CI setup by automatically identifying flaky tasks (e2e tests in particular) and re-running them
+
+## How?
+
+Nx Cloud follows a task-based CI model which is **not just fast but also robust**. Instead of statically assigning work to machines (as in a traditional CI model), you can imagine a **pile of tasks which get picked up automatically** by agents on the Nx Cloud infrastructure. There's **no static assignment, agents coordinate the work by themselves**. If an agent fails during the setup phase, other agents will pick up its work. If more work needs to be done, more agents will be started to still guarantee a fast execution, if less work is needed, fewer agents will be started to save resources.
+
+All this is possible because Nx Cloud directly integrates with Nx and has knowledge about the project structure and tasks as well as dependencies among them.
+
+Read more about individual features of Nx Cloud in the [features section](/ci/features).
+
+## How does Nx Cloud integrate into my existing CI setup
+
+Nx Cloud directly integrates with your existing CI setup.
+
+![Nx Cloud Overview](/shared/images/nx-cloud/nx-cloud-overview.webp)
+
+In a nutshell, here's how this works:
+
+**Step 1: Connect your workspace to Nx Cloud**
+
+This can be done by signing up on [nx.app](https://nx.app) and then connecting to your git repository. Read more about all the details [on this doc page](/ci/recipes/set-up/connect-to-cloud).
+
+**Step 2: Your CI script triggers Nx Cloud**
+
+```yml
+- name: Start CI run
+  run: 'npx nx-cloud start-ci-run --distribute-on="8 linux-medium-js"'
+```
+
+Check out our [recipes for the various CI providers](/ci/recipes/set-up).
+
+**Step 3: Run your Nx command as usual**
+
+```yml
+- run: npx nx-cloud record -- nx format:check
+- run: npx nx affected -t lint test build e2e-ci
+```
+
+All these commands are automatically picked up by Nx Cloud, split up into smaller tasks and distributed across the specified number of machines.
+
+**Step 4: All results are played back automatically**
+
+Nx Cloud automatically plays back all results to your CI system, as if distribution never happened. You can continue doing post-processing on the results, like uploading test reports, deploying artifacts etc.
```

Comment:
This section is no longer talking about "Why Nx Cloud?" and it is duplicating content that already exists in several places.

I would rewrite it to talk about the mental model of writing your CI pipeline as if it were executing on your local machine, but Nx automatically makes it faster for you.  Then you can talk about how that process works at each step.

So you can keep some of this same content - but don't frame it with `Step 1`, `Step 2` etc.

- To start using cloud, there's a simple registration process. (Link to recipes)
- Nx Cloud makes distribution across machines easy. Changing the way tasks are distributed is controlled with a single command.
- The rest of the CI pipeline runs the same commands you would run on your local machine.  This allows you to think about the pipeline as if it were run locally, but Nx allows it to be efficiently parallelized when it is run in CI.
- This is made possible because Nx Cloud automatically plays back all results to your CI system, as if distribution never happened.

Reply:
I'm aware there's duplication, but I think that's fine. I intentionally made it as little as possible, but I want to give the full picture here. I had only such bullet points first, but I don't want to have ppl jump again to pages etc, but at least get to see "that single line" etc. Also seeing some concrete things is easier to parse than just bullet points.

If they then go and implement Nx Cloud they'd have to dig deeper anyway, so this page is to give an overview and get the bird-view understanding of what's going on.